### DISCUSSION TITLE
How to use in Next js?

### DISCUSSION THREAD

**Discussion by waleedrana777** [themesberg/flowbite-react#80] [Q&A] [answered]
How to import in next js? I have tried including `import "flowbite";` in my _app.js file or in my component file but doesn't work!

**Comment by tulup-conner** [comment, marked as answer]
Hey there, a couple of things:

1. You don't actually need to add `import "flowbite"` anymore; we will update the README to reflect this, and it may be causing this issue for you (not sure)
2. If that isn't the cause of the problem, could you kindly provide a Github repo that reproduces this issue? We want to make sure Next.js users can use this software.

**Reply by waleedrana777** [reply]
Hi, I created this repo for testing: [Repo Link](https://github.com/waleedrana777/flowbitetest)
I followed all steps except last one with import flowbite in read me file! The javascript logic is not working for the accordian component! Can you make a basic tutorial for next js? Would be really cool!

**Reply by bacali95** [reply]
@waleedrana777 I see that you are using the raw `flowbite` components in your example, you need to use the `flowbite-react` components to have the interactivity work as expected for NextJs :slightly_smiling_face:

**Reply by waleedrana777** [reply]
Can you check out the repo now please, it works but the styles aren't applied for some reason even when I included the plugin!

**Reply by waleedrana777** [reply]
It look like this:
![image](https://user-images.githubusercontent.com/37755440/166891749-3bdc7b18-6d55-417b-ab5e-c8f386833f6f.png)

**Reply by bacali95** [reply]
Hello again, this can be fixed by adding `"node_modules/flowbite-react/**/*.{js,jsx,ts,tsx}"` to the `content` field in `tailwind.config.js`.

FYI we are gonna work on some tutorials and some templates for Next.js, Remix,... so it becomes more clear on how to use this library.

**Reply by waleedrana777** [reply]
That's awesome. It works now! I had already tried this but I was hardcoding the index.js in this folder! 

A tutorial is what I am hoping for. It would be a good start for this awesome tool! Most people don't know about flow-bite, I accidently found out through search! Thank you for your valuable time!

**Reply by waleedrana777** [reply]
![image](https://user-images.githubusercontent.com/37755440/166893645-f3b9c16f-78f4-4169-abd6-8520bf31278d.png)

**Reply by bacali95** [reply]
Always happy to help. you can join our [Discord server](https://discord.gg/4eeurUVvTy) to get the latest updates about this library and Flowbite in general. have a nice day :smile:

**Reply by xiaolin911211** [reply]
> It look like this: ![image](https://user-images.githubusercontent.com/37755440/166891749-3bdc7b18-6d55-417b-ab5e-c8f386833f6f.png)

Do you have a sample, because i am getting ReferenceError: document is not defined error with Nextjs

**Reply by bacali95** [reply]
Hello @xiaolin911211, we are currently having a bug within the library that breaks the support with Nextjs, I will push/publish a fix soon,

**Reply by bacali95** [reply]
Can you try `flowbite-react@0.0.25`

**Reply by xiaolin911211** [reply]
just tried, no effect when using 0.0.25 version.

**Comment by tulup-conner** [comment]
Following up, I just created a starter for `flowbite` + `next.js`: https://github.com/tulup-conner/flowbite-next-starter

**Comment by SutuSebastian** [comment]
Please read the [official documentation for Next.js](https://flowbite-react.com/docs/guides/nextjs).
